publish_courses/tests/test_api.py

```diff
@@ -1,125 +1,199 @@
-        self.request.query_params = query_params
-
-        summary_list = api.SummaryList(request=self.request)
-        return summary_list.list(self.request)
-
-    # Tests for BulkPublishListCreate:
-    def test_bulk_publish_list_create_success(self):
+    def test_returns_success(self):
         """
-        returns data as expected on success
+        returns expected data and status code on success
         """
-        data = {
-            'account': 'abc',
-            'term': '2015-1'
-        }
-        response = self._prep_request_and_post(data)
-        # validate data and status code on success
-        self.assertEqual(response.status_code, 201)
+        response = self._prep_request_and_post()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
         self.assertEqual(response.data['state'], 'queued')
         self.assertNotEquals(response.data['date_created'], None)
 
-    def test_bulk_publish_list_create_invalid_account(self):
+    def test_queues_job(self):
         """
-        invalid account raises permission denied
+        queues job on success
         """
-        data = {
-            'account': 'someaccount',
-            'term': '2015-1'
-        }
-        with self.assertRaises(PermissionDenied):
-            self._prep_request_and_post(data)
+        queue = django_rq.get_queue(QUEUE_NAME)
+        self.assertEqual(queue.count, 0)
+
+        response = self._prep_request_and_post()
 
-    @unittest.skip('BulkPublishListCreate seems to be bypassing its permission_classes')
-    # all of our unit tests should be failing, because the LTIPermission isn't
-    # being explicitly stubbed out, but they are passing, and this one is
-    # failing; perhaps switching to using the live server will trigger the
-    # permissions classes, or perhaps not mocking out the User object?...
-    @patch('publish_courses.api.LTIPermission.has_permission')
-    def test_bulk_publish_list_create_invalid_lti_session(self, mock_lti_check):
+        self.assertEqual(queue.count, 1)
+
+    def test_invalid_account(self):
         """
-        invalid lti session raises permission denied
+        invalid account raises permission denied
         """
         data = {
-            'account': 'abc',
+            'account': 'someaccount',
             'term': '2015-1'
         }
-        mock_lti_check.return_value = False
         with self.assertRaises(PermissionDenied):
             self._prep_request_and_post(data)
 
-    def test_bulk_publish_list_create_missing_data(self):
+    def test_missing_data(self):
         """
         Invalid/missing data raises Exception (REST framework converts that
-        into a status code 500)
+        into a response with status code 400)
         """
         data = {}
-        with self.assertRaises(Exception):
+        msg = 'account and term are required'
+        with self.assertRaisesRegexp(DRFValidationError, msg):
             self._prep_request_and_post(data)
```

Comment:
Yes, good call, it also guards against future updates to the method signature that might add positional args.